src/olympia/search/filters.py

```diff
@@ -707,7 +707,7 @@ def primary_should_rules(self, search_query, lang):
                             'match': {
                                 'name.trigrams': {
                                     'query': search_query,
-                                    'minimum_should_match': '66%',
+                                    'minimum_should_match': '67%',
```

Comment:
`3*66/100 = 1.98`
`3*67/100 = 2.01`

Since 5 character words like "Adobe" have 3 trigrams, and `minimum_should_match` is always rounded down to the lowest integer, that change is enough to fix the problem and ensure "Dobromat" doesn't match "Adobe".